robusta_krr/core/integrations/prometheus/metrics_service/prometheus_metrics_service.py

```diff
@@ -175,11 +175,41 @@ async def gather_data(
         self.debug(f"Gathering data for {object} and {resource}")
 
         await self.add_historic_pods(object, period)
-
+        if resource == ResourceType.Memory:
+            await self.get_oomkill_count(object)
         MetricLoaderType = BaseMetricLoader.get_by_resource(resource)
         metric_loader = MetricLoaderType(self.config, self.prometheus)
         return await metric_loader.load_data(object, period, step, self.name())
 
+    async def get_oomkill_count(
+        self, object: K8sObjectData, period: datetime.timedelta = datetime.timedelta(days=1)
+    ) -> None:
+        """
+        Counts the relevant (limit related) oomkills in designated period
+        Args:
+            object (K8sObjectData): The Kubernetes object.
+            period (timedelta): The period to query, by default the past day.
+        """
+        # no limit so no oomkill isnt related to limit
+        if not object.allocations.limits[ResourceType.Memory]:
+            return
+        pods_selector = "|".join(pod.name for pod in object.pods)
```

Comment:
This one could result in performance issues. Also this will be changed in soon version.
We should think on how to optimize it with PromQL joins